driver/src/price_estimation/mod.rs

```diff
@@ -32,19 +37,29 @@ pub struct PriceOracle {
     /// whitelisted tokens get their prices estimated.
     tokens: TokenData,
     /// The price source to use.
-    ///
-    /// Note that currently only one price source is supported, but more price
-    /// source are expected to be added in the future.
     source: Box<dyn PriceSource>,
 }
 
 impl PriceOracle {
     /// Creates a new price oracle from a token whitelist data.
-    pub fn new(tokens: TokenData) -> Result<Self> {
-        Ok(PriceOracle::with_source(
-            tokens,
-            AveragePriceSource::new(KrakenClient::new()?, DexagClient::new()?),
-        ))
+    pub fn new(tokens: TokenData, update_interval: Duration) -> Result<Self> {
+        Ok(if tokens.is_empty() {
+            // If there are no tokens we don't need to create a real price
+            // source. This avoids sending requests in the Dexag constructor.
```

Comment:
Looking at this comment now, I think it's weird to have the constructor perform a long running operation (they are usually non-blocking). Maybe not super critical now, but I'd prefer if we can move the token initialisation to be lazy.

Reply:
I implemented this in https://github.com/gnosis/dex-services/pull/626 .